### PR TITLE
basic NetBSD platform support

### PR DESCRIPTION
**Pull request by justincormack**
This gets to the point where it tries to build, will work on from there once I have a fixed clang C++ setup.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -113,6 +113,10 @@ case $OSName in
         __BuildOS=OpenBSD
         ;;
 
+    NetBSD)
+	__BuildOS=NetBSD
+	;;
```
Comment (akoeplinger) [inline build.sh]: indentation seems off here (tabs vs spaces?)

**Comment by justincormack** [comment]
Ok, fixed the whitespace issues.

**Comment by kangaroo** [comment]
LGTM

**Comment by janvorli** [comment]
LGTM

**Pull request merged**